net/dns/resolvconfpath_default.go

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !gokrazy
```

Comment:
We don't use this build tag anywhere else and this isn't a GOOS.

Let's do this at runtime with a comparison to distro.Get() == distro.Gokrazy instead.

Reply:
If you want to minimize how often/ugly that looks doing it at runtime, you can do it just in the DNS manager constructor and add new fields for these two strings.

Reply:
Concrete counter-proposal: #8690

Reply:
I guess the gokrazy build tool always sets the `gokrazy` tag, huh? So maybe this PR is fine.

I think I have a slight preference for not using build tags if we can avoid it, though.

Reply:
But let's get your change in first. LGTM if it works for gokr.